lib/Cake/Network/Email/CakeEmail.php

```diff
@@ -353,6 +353,10 @@ public function __construct($config = null) {
 
 		if ($config) {
 			$this->config($config);
+		} elseif (config('email')) {
```

Comment:
[_applyConfig()](https://github.com/cakephp/cakephp/blob/master/lib/Cake/Network/Email/CakeEmail.php#L1226) will call `config('email')` again to read the config file. To avoid that it might be better to create an instance of `EmailConfig` and store in protected property here itself, which `_applyConfig()` can reuse.